client/src/plugins/usage-statistics/__tests__/UsageStatisticsSpec.js

```diff
@@ -0,0 +1,397 @@
+  });
+
+
+  it('should enable when mounted', async () => {
+
+    // given
+    Flags.init({
+      [ ET_ENDPOINT ]: 'test-et-endpoint'
+    });
+
+    const { instance } = createUsageStatistics({
+      configValues: { 'editor.privacyPreferences': { ENABLE_USAGE_STATISTICS: true } }
+    });
+
+    const enableSpy = sinon.spy();
+
+    instance.enable = enableSpy;
+
+    // when
+    await instance.componentDidMount();
+
+    // then
+    expect(enableSpy).to.have.been.called;
+  });
+
+
+  it('should enable registered event handlers', async () => {
+
+    // given
+    Flags.init({
+      [ ET_ENDPOINT ]: 'test-et-endpoint'
+    });
+
+    const { instance } = createUsageStatistics({
+      configValues: { 'editor.privacyPreferences': { ENABLE_USAGE_STATISTICS: true } }
+    });
+
+    const eventHandler = new BaseEventHandler('test-event', () => {});
+
+    instance._eventHandlers = [ eventHandler ];
+
+    // when
+    await instance.componentDidMount();
+
+    // then
+    expect(eventHandler.isEnabled()).to.eql(true);
+  });
+
+
+  it('should disable registered event handlers', async () => {
+
+    // given
+    Flags.init({
+      [ ET_ENDPOINT ]: 'test-et-endpoint'
+    });
+
+    const { instance } = createUsageStatistics({
+      configValues: { 'editor.privacyPreferences': { ENABLE_USAGE_STATISTICS: true } }
+    });
+
+    const eventHandler = new BaseEventHandler('test-event', () => {});
+
+    instance._eventHandlers = [ eventHandler ];
+
+    // when
+    await instance.componentDidMount();
+
+    instance.disable();
+
+    // then
+    expect(eventHandler.isEnabled()).to.eql(false);
+  });
+
+
+  it('should listen to privacy preference change event', async () => {
+
+    // given
+    Flags.init({
+      [ ET_ENDPOINT ]: 'test-et-endpoint'
+    });
+
+    const subscribeSpy = sinon.spy();
+
+    const { instance } = createUsageStatistics({
+      configValues: { 'editor.privacyPreferences': { ENABLE_USAGE_STATISTICS: true } },
+      subscribeSpy
+    });
+
+    instance.fetch = () => {};
+
+    // when
+    await instance.componentDidMount();
+
+    // then
+    expect(subscribeSpy).to.have.been.calledWith('privacy-preferences.changed');
+  });
+
+
+  it('should enable when usage statistics preference switched on', async () => {
+
+    // given
+    Flags.init({
+      [ ET_ENDPOINT ]: 'test-et-endpoint'
+    });
+
+    const subscribeSpy = sinon.spy();
+
+    let usageStatisticsEnabled = false;
+
+    const { instance } = createUsageStatistics({
+      configGet: () => {
+        return { ENABLE_USAGE_STATISTICS: usageStatisticsEnabled };
+      },
+      subscribeSpy
+    });
+
+    const enableSpy = sinon.spy();
+    instance.enable = enableSpy;
+
+    // when
+    await instance.componentDidMount();
+
+    // then
+    expect(enableSpy).to.not.have.been.called;
+
+    // when
+    const fn = getSubscriptionCallbackFromSpy(subscribeSpy, 'privacy-preferences.changed');
+
+    usageStatisticsEnabled = true;
```

Comment:
This does not really do anything, cf.
```bash
▶ node
Welcome to Node.js v12.16.1.
Type ".help" for more information.
> let boolean = true
undefined
> const obj = { boolean: boolean }
undefined
> obj
{ boolean: true }
> boolean = false
false
> obj
{ boolean: true }
>
```

If you want to change object property, you need to grab the object itself.

Reply:
Well becaus in the example you provided you don't construct a new object but you re-print its reference. In the code block we constrcut a new object and return it. So when you call the function again, it returns the updated value.

Reply:
```javascript
let val = false;

var fn = function(){
  return val;
}

fn(); // false;

val = true;
fn(); // true;
```

Reply:
True